### ISSUE TITLE
In Editor, suggest with priority the entities already used in Annotator for this span; in Annotator, suggest entities used in Statement modelling this text span

### ISSUE THREAD

**Issue by davidzbiral** [DISSINET/InkVisitor#2033]
A very realistic, if not major, workflow with the annotator in place will be:
1. I delimit the statement.
2. I identify entities.
3. Only then I structure them, i.e. put them as actants, property components, etc.

We want to give the user both options: (A) the approach "first make the statement, then anchor entities in full-text", and (B) "first identify entities, then structure the statement. In both cases, we want that there is pre-suggesting assistance, prioritizing in the suggester in Annotator (strategy A above) and Editor (strategy B) the entities already identified in the other component. I.e. making suggestions efficient - already when clicking on the empty suggester, suggesting immediately the entities used in this statement (strategy A) or in this span's annotation (strategy B).

**Comment by adammertel** [comment]
This is a good candidate for a new suggester strategy, added a new context here 
https://docs.google.com/document/d/1VM5WMQnuyRlr2DzTYO9YUk7K6Y2cFZiwwc2HhC3BZ_U/edit

**Comment by Ptrhnk** [comment]
From the Slack conversation: Suggest entities in Editor from Statement first then from nearest subT.

**Comment by davidzbiral** [comment]
Yes - of course you need to put some threshold on the number of suggestions, but you should start from the closest (statement where I am), subT I am in, then following the T tree up. In addition, once the text is segmented, and thus have T anchors, the T anchors per se are not interesting as suggestions, so deprioritize them so that they are at the very end of suggestions (if at all).

**Comment by ReimaValimaki** [comment]
I would say T anchors are not at all relevant for suggestions. With few exceptions, T anchors are created once when segmenting the text.

**Comment by davidzbiral** [comment]
Yes, exactly - this is what I meant too - so Ts tree is followed up to prioritize suggestions from closest to furthest context, but T tags are not relevant as suggestions themselves.
